### ISSUE TITLE
Allmusic.com Tag Source fails with displayed Unhandled Error

### ISSUE THREAD

**Issue by bernd-wechner** [puddletag/puddletag#666]
Displayed under the search box:

![image](https://user-images.githubusercontent.com/7296506/149701975-46efc870-afc3-4196-978a-ac96ba5d8d11.png)

That is using the github clone (as it's not in the Mint repos).

**Comment by sandrotosi** [comment]
@bernd-wechner can you try running puddletag in a terminal and see if there's a traceback you can cut-and-paste here? even an absence of it would be useful to know.

can you report if a specific file was the cause of the issue, and is so what the filename/tag is?

thanks

**Comment by bernd-wechner** [comment]
Here's the stack trace on the console:

```
puddletag Version: 2.0.1, Changeset: e5fb85c34ff2e699ca35274298d1921f2598be98
Locale: en_AU
Traceback (most recent call last):
  File "/media/Data/Workspace/Eclipse/puddletag/source/puddlestuff/webdb.py", line 866, in search
    return tag_source_search(self.curSource, group, files)
  File "/media/Data/Workspace/Eclipse/puddletag/source/puddlestuff/webdb.py", line 557, in tag_source_search
    albums = ts.search(primary, group[primary])
  File "/media/Data/Workspace/Eclipse/puddletag/source/puddlestuff/tagsources/amg.py", line 616, in search
    search_results = parse_searchpage(searchpage, artist, album)
  File "/media/Data/Workspace/Eclipse/puddletag/source/puddlestuff/tagsources/amg.py", line 363, in parse_searchpage
    page = get_encoding(page, True, 'utf8')[1]
  File "/media/Data/Workspace/Eclipse/puddletag/source/puddlestuff/tagsources/__init__.py", line 59, in get_encoding
    match = re.search('<\?xml(.+)\?>', page)
  File "/usr/lib/python3.8/re.py", line 201, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a string pattern on a bytes-like object
```

I've seen this string/bytes issue before and it's possible a library version issue (as one library or another moves from taking of byte strings to new strings or vice versa or whatever). re.py in this instance. 

So I did the right thing, created a new venv, activated it and installed from requirements.txt:

```
python -m venv ~/.virtualenvs/puddletag
source ~/.virtualenvs/puddletag/bin/activate
pip install -r requirements.txt 
```
And I run puddletag and AllMusic no longer appears on the list of possible Tag Sources. Wow. So I deactivate the venv and run puddletag and it does appear and produces that error when used. It's independent of the files selected.

Looks like I might set it up in my debugger and try and work it out when I find time. 

I have of course followed all the instructions in the README:

https://github.com/puddletag/puddletag#installation 

under Source code.

As an aside: I used mp3tag for years and loved it and then puddletag for years and loved it, but it was stagnant on Python2  for a while and my music processing went into hiatus for some years (children). I had the whim to look at my music once more and checked out puddletag and was blown away and pleased that it's now Python3 and QT5 and which means among other things I'm more than happy to open it up in my IDE and start looking at the code and diagnosing stuff myself and fixing it (there are some intermittent hard to reproduce bugs, albeit easy to reproduce when found thanks to the undo feature and reapplying - mostly actions I have that do some odd stuff like erasing field unexpectedly now and then, have no clear steps to reproduce so not worth opening an issue yet - would have to catch it in action and drill down). In any case thanks immensely for the work on this, best tag editors I have ever used (mp3tag/puddletag) just so well designed compared to most I look at with intuitive work flows and lots of configurability).

PPS: I'd love to solve this: https://github.com/puddletag/puddletag/issues/664 ;-). I guess once I have it working in an IDE I can drill into that too.
